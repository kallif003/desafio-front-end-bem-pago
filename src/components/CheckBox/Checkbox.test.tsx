import * as React from "react"
import { render } from "@testing-library/react"
import userEvent from "@testing-library/user-event"

import { CheckboxDiv, CheckboxContainer, Checkbox, TextCheckBox } from "./"

describe("Checkbox components", () => {
	it("Renders CheckboxDiv properly", () => {
		const { container } = render(<CheckboxDiv>Test</CheckboxDiv>)
		expect(container.firstChild).toMatchSnapshot()
	})
	it("Renders CheckboxContainer properly", () => {
		const { container } = render(<CheckboxContainer>Test</CheckboxContainer>)
		expect(container.firstChild).toMatchSnapshot()
	})
	it("Renders Checkbox properly", () => {
		const { container } = render(<Checkbox>Test</Checkbox>)
		expect(container.firstChild).toMatchSnapshot()
	})
	it("Renders TextCheckBox properly", () => {
		const { container } = render(<TextCheckBox>Test</TextCheckBox>)
		expect(container.firstChild).toMatchSnapshot()
	})
	it("test Checkbox button ", () => {
		const { container } = render(<Checkbox>Test</Checkbox>)
		userEvent.click(container)
	})
})
