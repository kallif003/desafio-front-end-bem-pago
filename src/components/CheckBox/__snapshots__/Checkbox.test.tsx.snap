// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox components Renders CheckboxContainer properly 1`] = `
<div
  class="sc-gsDKAQ DUhMA"
>
  Test
</div>
`;

exports[`Checkbox components Renders CheckboxDiv properly 1`] = `
<div
  class="sc-bdvvtL hLVIjI"
>
  Test
</div>
`;

exports[`Checkbox components Renders TextCheckBox properly 1`] = `
<label
  class="sc-hKwDye gVnGAI"
>
  Test
</label>
`;
